packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,510 @@
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("bufferingUpdate")},
+        {flutter::EncodableValue("values"), flutter::EncodableValue(position)}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer.onBuffering] send bufferingUpdate event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::SendBufferingEnd() {
+  if (event_sink_) {
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("bufferingEnd")}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer.onBuffering] send bufferingEnd event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::OnPrepared(void *data) {
+  VideoPlayer *player = (VideoPlayer *)data;
+  LOG_INFO("[VideoPlayer.onPrepared] video player is prepared");
+  if (!player->is_initialized_) {
+    player->SendInitialized();
+  }
+}
+
+void VideoPlayer::OnBuffering(int percent, void *data) {
+  LOG_INFO("[VideoPlayer.onBuffering] percent: %d", percent);
+  VideoPlayer *player = (VideoPlayer *)data;
+  if (percent == 100) {
+    player->SendBufferingEnd();
+    player->is_buffering_ = false;
+  } else if (!player->is_buffering_ && percent <= 5) {
+    player->SendBufferingStart();
+    player->is_buffering_ = true;
+  } else {
+    player->SendBufferingUpdate(percent);
+  }
+}
+
+void VideoPlayer::OnSeekCompleted(void *data) {
+  VideoPlayer *player = (VideoPlayer *)data;
+  LOG_INFO("[VideoPlayer.onSeekCompleted] completed to seek");
+  if (player->on_seek_completed_) {
+    player->on_seek_completed_();
+    player->on_seek_completed_ = nullptr;
+  }
+}
+
+void VideoPlayer::OnPlayCompleted(void *data) {
+  VideoPlayer *player = (VideoPlayer *)data;
+  LOG_INFO("[VideoPlayer.onPlayCompleted] completed to play video");
+  if (player->event_sink_) {
+    flutter::EncodableMap encodables = {{flutter::EncodableValue("event"),
+                                         flutter::EncodableValue("completed")}};
+    flutter::EncodableValue eventValue(encodables);
+    player->event_sink_->Success(eventValue);
+    LOG_DEBUG("[VideoPlayer.onPlayCompleted] change player state to pause");
+    player->Pause();
+  }
+}
+
+void VideoPlayer::OnError(int error_code, void *user_data) {
+  VideoPlayer *player = (VideoPlayer *)user_data;
+  LOG_ERROR("[VideoPlayer.onError] error code: %s",
+            get_error_message(error_code));
+  if (player->event_sink_) {
+    player->event_sink_->Error("Video player had error",
+                               get_error_message(error_code));
+  }
+}
+
+void VideoPlayer::onInterrupted(player_interrupted_code_e code, void *data) {
+  VideoPlayer *player = (VideoPlayer *)data;
+  LOG_ERROR("[VideoPlayer.onErrorOccurred] error code: %s",
+            get_error_message(code));
+  if (player->event_sink_) {
+    player->event_sink_->Error("Video player had error",
```

Comment:
I think the error message could be more specific. For example "interrupted error"?

Reply:
> I think the error message could be more specific. For example "interrupted error"?

have fixed